File: microservices/api-core/internal/models/report.go
package models

import (
	"database/sql"
	"time"
)

type Report struct {
	ID          int           `json:"id"`
	ReporterID  int           `json:"reporter_id"`
	ReportedID  int           `json:"reported_id"`
	RequestID   sql.NullInt64 `json:"request_id,omitempty"`
	Reason      string        `json:"reason"` // 'spam', 'inappropriate', 'fraud', 'harassment', 'other'
	Description string        `json:"description"`
	Status      string        `json:"status"` // 'pending', 'reviewed', 'resolved', 'dismissed'
	CreatedAt   time.Time     `json:"created_at"`
	UpdatedAt   time.Time     `json:"updated_at"`
}

type CreateReportRequest struct {
	ReportedID  int    `json:"reported_id"`
	RequestID   *int   `json:"request_id,omitempty"`
	Reason      string `json:"reason"`
	Description string `json:"description"`
}

type ReportResponse struct {
	ID          int       `json:"id"`
	ReporterID  int       `json:"reporter_id"`
	ReportedID  int       `json:"reported_id"`
	RequestID   *int      `json:"request_id,omitempty"`
	Reason      string    `json:"reason"`
	Description string    `json:"description"`
	Status      string    `json:"status"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

func (r *Report) ToResponse() ReportResponse {
	resp := ReportResponse{
		ID:          r.ID,
		ReporterID:  r.ReporterID,
		ReportedID:  r.ReportedID,
		Reason:      r.Reason,
		Description: r.Description,
		Status:      r.Status,
		CreatedAt:   r.CreatedAt,
		UpdatedAt:   r.UpdatedAt,
	}

	if r.RequestID.Valid {
		rid := int(r.RequestID.Int64)
		resp.RequestID = &rid
	}

	return resp
}

File: mobile/lib/models/report.dart
class Report {
  final int id;
  final int reporterId;
  final int reportedId;
  final int? requestId;
  final String reason;
  final String description;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;

  Report({
    required this.id,
    required this.reporterId,
    required this.reportedId,
    this.requestId,
    required this.reason,
    required this.description,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'],
      reporterId: json['reporter_id'],
      reportedId: json['reported_id'],
      requestId: json['request_id'],
      reason: json['reason'],
      description: json['description'],
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'reporter_id': reporterId,
      'reported_id': reportedId,
      'request_id': requestId,
      'reason': reason,
      'description': description,
      'status': status,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }

  String get reasonDisplay {
    switch (reason) {
      case 'spam':
        return 'Spam';
      case 'inappropriate':
        return 'Contenido inapropiado';
      case 'fraud':
        return 'Fraude';
      case 'harassment':
        return 'Acoso';
      case 'other':
        return 'Otro';
      default:
        return reason;
    }
  }

  String get statusDisplay {
    switch (status) {
      case 'pending':
        return 'Pendiente';
      case 'reviewed':
        return 'Revisado';
      case 'resolved':
        return 'Resuelto';
      case 'dismissed':
        return 'Descartado';
      default:
        return status;
    }
  }
}

class CreateReportRequest {
  final int reportedId;
  final int? requestId;
  final String reason;
  final String description;

  CreateReportRequest({
    required this.reportedId,
    this.requestId,
    required this.reason,
    required this.description,
  });

  Map<String, dynamic> toJson() {
    return {
      'reported_id': reportedId,
      if (requestId != null) 'request_id': requestId,
      'reason': reason,
      'description': description,
    };
  }
}
